internal/proxy/proxy.go

```diff
@@ -83,6 +83,10 @@ func proxyHandler(logger hclog.Logger, client *vault.Client, cache *Cache) func(
 			}
 		}
 
+		// add user agent header
+		ua := config.GetUserAgentBase(client.Name, client.Version)
+		fwReq.Header.Set("User-Agent", ua+"; requesting from proxy")
```

Comment:
Is there a test we could add in proxy_test.go to test this is getting set as we expect?